src/tensor_operations/matrix_decomposition.jl

```diff
@@ -590,6 +590,27 @@ function sqrt_decomp(D::ITensor, u::Index, v::Index)
   return sqrtDL, prime(δᵤᵥ), sqrtDR
 end
 
+# Take the square root of T assuming it is Hermitian
+# TODO: add more general index structures
+function Base.sqrt(T::ITensor; ishermitian=true, atol=1e-15)
+  @assert ishermitian
+  # TODO diagonal version
+  #if isdiag(T) && order(T) == 2
+  #  return itensor(sqrt(tensor(T)))
+  #end
+  D, U = eigen(T; ishermitian=ishermitian)
+  sqrtD = D
+  for n in 1:mindim(D)
+    Dnn = D[n, n]
+    if Dnn < 0 && abs(Dnn) < atol
+      sqrtD[n, n] = 0
+    else
+      sqrtD[n, n] = sqrt(Dnn)
+    end
```

Comment:
Unfortunately this won't be good on GPU (and even on CPU will have bad efficiency since it is performing scalar indexing operations on ITensors).